app/controllers/application_controller.rb

```diff
@@ -17,6 +17,14 @@ def configure_permitted_parameters
     devise_parameter_sanitizer.for(:sign_up) << :first_name << :last_name
   end
 
+  rescue_from CanCan::AccessDenied do |exception|
+    if user_signed_in?
+      redirect_to root_path, alert: exception.message
+    else
+      redirect_to new_user_session_path , alert: "Please sign up to access this page."
```

Comment:
It would be great if this could be in one of the text template files, instead of baked into the code, using the `en.yml` file in `config/locales`. (See `devise.en.yml` for more examples, and look at the devise source code to see how those locale files are used.)